sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample10_SecretReference.md

```diff
@@ -0,0 +1,50 @@
+# Create, Retrieve and Delete a Secret Reference
+
+The AppConfiguration service supports KeyVault Secret reference settings.
+Secret references are settings that follow specific JSON schema for the `Value`, and the `application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8` content type.
+The `Azure.Data.AppConfiguration` library provides a strongly-typed way of managing Secret references.
+This sample shows how to use the library to create, retrieve, and delete Secret references.
+
+## Create a Secret Reference
+
+To create a secret reference, use the `SecretReferenceConfigurationSetting` class:
+
+```C# Snippet:Sample_CreateSecretReference
+var secretReferenceSetting = new SecretReferenceConfigurationSetting("setting", new Uri("https://<keyvault_name>.vault.azure.net/secrets/<secret_name>"));
+```
+
+## Set a Secret Reference
+
+The `SecretReferenceConfigurationSetting` inherits from the `ConfigurationSetting` class and can be passed to any method that accepts the `ConfigurationSetting`.
+
+To set the secret reference, pass the instance to the `SetConfigurationSetting` method:
+
+```C# Snippet:Sample_SetSecretReference
+client.SetConfigurationSetting(secretReferenceSetting);
+```
+
+## Retrieve a Secret Reference
+
+You can use the `GetConfigurationSetting` method to retrieve the secret reference.
+
+If you'd like to get the secret value you would need to reference the [Azure.Security.KeyVault.Secrets](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Azure.Security.KeyVault.Secrets/README.md) library and use the `KeyVaultSecretIdentifier` to parse the `SecretId`, and `SecretClient.GetSecretAsync` to get the secret value.
```

Comment:
Do you think a user will care which version of KV.Secrets this sample uses? I like it this way so we don't have to manually maintain a reference here. At the same time wonder if a user will care and will be able to find the version reference in our repo.

Reply:
I'll add a comment about when the  `KeyVaultSecretIdentifier` feature was added.